collect_app/src/main/java/org/odk/collect/android/instancemanagement/autosend/InstanceAutoSendFetcher.kt

```diff
@@ -1,23 +1,32 @@
 package org.odk.collect.android.instancemanagement.autosend
 
+import android.app.Application
 import org.odk.collect.forms.FormsRepository
 import org.odk.collect.forms.instances.Instance
 import org.odk.collect.forms.instances.InstancesRepository
+import org.odk.collect.settings.SettingsProvider
+import org.odk.collect.settings.enums.AutoSend
+import org.odk.collect.settings.enums.StringIdEnumUtils.getAutoSend
 
 object InstanceAutoSendFetcher {
 
     fun getInstancesToAutoSend(
+        application: Application,
         instancesRepository: InstancesRepository,
-        formsRepository: FormsRepository
+        formsRepository: FormsRepository,
+        settingsProvider: SettingsProvider
     ): List<Instance> {
         val allFinalizedForms = instancesRepository.getAllByStatus(
             Instance.STATUS_COMPLETE,
             Instance.STATUS_SUBMISSION_FAILED
         )
 
+        val autoSendSetting =
+            settingsProvider.getUnprotectedSettings().getAutoSend(application)
```

Comment:
We can probably flip how these enums work in the future so that the source of truth is in Kotlin/Java strings rather than in Android ones so we don't have to deal with `Context` in places like this. We'd just need to set the entry values for the UI preference programmatically instead. I think that's a change for another day however.